yt_dlp/extractor/vrt.py

```diff
@@ -176,156 +189,257 @@ def _real_extract(self, url):
+            display_id,
+            'Downloading asset JSON',
+            'Unable to download asset JSON',
+            data=json.dumps({
+                'operationName': 'VideoPage',
+                'query': self._VIDEO_PAGE_QUERY,
+                'variables': {
+                    'pageId': urllib.parse.urlparse(url).path,
+                },
+            }).encode(),
+            headers={
+                'content-type': 'application/json',
+                'x-vrt-client-name': 'WEB',
+                'x-vrt-client-version': '1.5.9',
+                'x-vrt-zone': 'default',
+            },
+        )['data']['page']
+
+        video_id = metadata['player']['modes'][0]['streamId']
+        video_token = self._fetch_video_token()
+
+        try:
+            streaming_info = self._call_api(video_id, 'vrtnu-web@PROD', id_token=video_token)
+        except ExtractorError as e:
+            if not video_token and isinstance(e.cause, HTTPError) and e.cause.status == 404:
+                self.raise_login_required()
+            raise
+
+        formats, subtitles = self._extract_formats_and_subtitles(streaming_info, video_id)
+
+        code = traverse_obj(streaming_info, ('code', {str}))
+        if not formats and code:
+            if code in ('CONTENT_AVAILABLE_ONLY_FOR_BE_RESIDENTS', 'CONTENT_AVAILABLE_ONLY_IN_BE', 'CONTENT_UNAVAILABLE_VIA_PROXY'):
+                self.raise_geo_restricted()
+            elif code in ('CONTENT_AVAILABLE_ONLY_FOR_BE_RESIDENTS_AND_EXPATS', 'CONTENT_IS_AGE_RESTRICTED', 'CONTENT_REQUIRES_AUTHENTICATION'):
+                self.raise_login_required()
+            else:
+                raise ExtractorError(f'Unable to extract formats: {code}')
 
         return {
-            **traverse_obj(details, {
-                'title': 'title',
-                'description': ('description', {clean_html}),
-                'timestamp': ('data', 'episode', 'onTime', 'raw', {parse_iso8601}),
-                'release_timestamp': ('data', 'episode', 'onTime', 'raw', {parse_iso8601}),
-                'series': ('data', 'program', 'title'),
-                'season': ('data', 'season', 'title', 'value'),
-                'season_number': ('data', 'season', 'title', 'raw', {int_or_none}),
-                'season_id': ('data', 'season', 'id', {str_or_none}),
-                'episode': ('data', 'episode', 'number', 'value', {str_or_none}),
-                'episode_number': ('data', 'episode', 'number', 'raw', {int_or_none}),
-                'episode_id': ('data', 'episode', 'id', {str_or_none}),
-                'age_limit': ('data', 'episode', 'age', 'raw', {parse_age_limit}),
-            }),
-            'id': video_id,
-            'display_id': display_id,
+            **self._json_ld(
+                traverse_obj(metadata, ('ldjson', ..., {json.loads})),
+                video_id,
+                fatal=False,
+            ),
+            **traverse_obj(metadata, ('episode', {
```

Comment:
@bergoid's version also had `release_date`, `release_timestamp` & `upload_date`, so maybe we should add these as well?

  ```py
**traverse_obj(metadata, {
    'release_date': ('episode', 'onTimeRaw', {unified_strdate}),
    'release_timestamp': ('episode', 'onTimeRaw', {parse_iso8601}),
    'upload_date': ('episode', 'onTimeRaw', {unified_strdate}),
})
```